File: src/acquire/gui/MouseUtil.java
package acquire.gui;

public class MouseUtil {
    static double mouseX;
    static double mouseY;
    static double clickX;
    static double clickY;
    static boolean isMouseInScene;
    static boolean isMouseClicked;
    static boolean isMousePressed;

    static void setPosition(double x, double y) {
        mouseX = x;
        mouseY = y;
    }

    static void setClickPosition(double x, double y) {
        clickX = x;
        clickY = y;
    }

    static boolean mouseOver(Actor actor) {
        return actor.boundary().contains(mouseX, mouseY);
    }
    static boolean mouseClicked(Actor actor) {
        if (isMouseClicked && actor.boundary().contains(clickX, clickY)) {
            isMouseClicked = false; // un-register the click
            return true;
        } else {
            return false;
        }
    }
}

File: src/theatre/core/MouseUtil.scala
package theatre.core

object MouseUtil {
  var mouseX: Double = .0
  var mouseY: Double = .0
  var clickX: Double = .0
  var clickY: Double = .0
  var isMouseInScene: Boolean = false
  var isMouseClicked: Boolean = false
  var isMousePressed: Boolean = false

  def setPosition(x: Double, y: Double) {
    mouseX = x
    mouseY = y
  }

  def setClickPosition(x: Double, y: Double) {
    clickX = x
    clickY = y
  }

  def mouseOver(actor: Actor): Boolean =
    actor.boundary.contains(mouseX, mouseY)

  def mouseClicked(actor: Actor): Boolean = {
    if (isMouseClicked && actor.boundary.contains(clickX, clickY)) {
      isMouseClicked = false
      true
    }
    else {
      false
    }
  }
}
